### PR TITLE
Add Python API and documentation for Queue/Stack Tables

### PR DESCRIPTION
**Pull request by willfindlay**
This pull request adds Python API support for #2993. It also adds documentation for BPF and Python APIs.

Changes:
- added QueueStack table type
- small changes to BPF.get_table() make QueueStack work properly
- added documentation
- added unit tests for Python API

### CODE REVIEW

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Comment by yonghong-song** [comment]
@s41m0n If you have some time, could you help review the patch? Thanks!

File: docs/reference_guide.md [hunk 1]
```diff
@@ -1051,7 +1059,45 @@ BPF_ARRAY(ex2, int, 1024);
 BPF_HASH_OF_MAPS(maps_hash, "ex1", 10);
 ```
 
-### 15. map.lookup()
+### 15. BPF_STACK
```
[the remark this answers is not in the document]
Comment (willfindlay) [reply docs/reference_guide.md:1062]: @s41m0n Actually I think it's 4.20. The 4.2 in the kernel version check is actually a mistake by me, thanks for pointing it out!

**Comment by willfindlay** [comment]
@yonghong-song Should be good to test again. I had the wrong kernel version check in my tests.

**Comment by willfindlay** [comment]
> It looks very good to me. I did not test it, but both methods and syntax seem compliant.
> I had a doubt about the kernel release where QUEUE/STACK have been introduced, so I checked: you are right, it is from v.4.2. So, if it is not a problem (I can open a different PR in case) in the test file I created for CC (https://github.com/iovisor/bcc/blob/master/tests/cc/test_queuestack_table.cc#L23) we should fix the kernel version check (my mistake, don't know but I was sure they were available only from v.5.0).
> Congrats!

Thanks for the comments. For the record, 4.2 was a mistake. It should be 4.20, when you fix your version check.

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Comment by s41m0n** [comment]
> > It looks very good to me. I did not test it, but both methods and syntax seem compliant.
> > I had a doubt about the kernel release where QUEUE/STACK have been introduced, so I checked: you are right, it is from v.4.2. So, if it is not a problem (I can open a different PR in case) in the test file I created for CC (https://github.com/iovisor/bcc/blob/master/tests/cc/test_queuestack_table.cc#L23) we should fix the kernel version check (my mistake, don't know but I was sure they were available only from v.5.0).
> > Congrats!
> 
> Thanks for the comments. For the record, 4.2 was a mistake. It should be 4.20 when you fix your version check.

Yes sure, so I'm gonna make a different PR or do you prefer to fix that kernel version check here @yonghong-song ?

**Comment by yonghong-song** [comment]
> Yes sure, so I'm gonna make a different PR or do you prefer to fix that kernel version check here @yonghong-song ?

@s41m0n Yes, the c++ test kernel version can be fixed in a different PR.

**Comment by yonghong-song** [comment]
LGTM. Thanks!

**Pull request merged**